server/lib/analytics/tick-timer.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Internal dependencies
+ */
+import analytics from './index';
+
+export function logTimingOfNextTick() {
+	/*
+	 * Call setImmediate once to enqueue a function at the end of the current
+	 * iteration of the event loop. From within the queued function, call
+	 * setImmedate a second time to schedule a callback to be called at the end
+	 * of the following event loop. As per the documentation, a setImmediate
+	 * callback scheduled within another setImmediate callback is guaranteed to
+	 * be run in the following iteration of the event loop.
+	 *
+	 * See:
+	 * https://nodejs.org/en/docs/guides/event-loop-timers-and-nexttick/
+	 * https://nodejs.org/api/timers.html#timers_setimmediate_callback_args
+	 */
+	setImmediate( () => {
+		const startTime = new Date();
```

Comment:
@ehg Yeah, that looks like the right tool for the job 👍 

I'm thinking that the extra resolution isn't necessary though. From my local testing, typical timing values are in the range of 50ms to 300ms, so I think reporting with millisecond precision is plenty. Although I suppose reporting in nanoseconds wouldn't hurt.

Thoughts?